### PR TITLE
Add deprecate to qiskit-ibmq-provider entry on the sidebar

### PR DESCRIPTION
**Pull request by HuangJunye**
### Summary

https://github.com/Qiskit/qiskit-ibmq-provider is now deprecated and replaced by https://github.com/Qiskit/qiskit-ibm-provider. We should reflect this change in documentation. #1638 will reinistate qiskit-ibmq-provider api reference entry to sidebar. We should label qiski-ibmq-provider deprecated.

Related: https://github.com/Qiskit/web-components/issues/196

cc: @kt474, @1ucian0 

### Details and comments

~To distinguish the two providers, I propose to name the old one `Qiskit IBMQ Provider` and the new one `Qiskit IBM Quanum Provider`.~

I realized later that the old provider name is `Qiskit IBM Quantum Provider` and the new one is `Qiskit IBM Provider`.

```rst
.. toctree::
   :caption: Other API References
   :hidden:

   Qiskit Aer <apidoc/aer>
   Qiskit IBM Quantum Provider (deprecated) <apidoc/ibmq-provider>
   Qiskit IBM Provider (new) <https://qiskit.org/documentation/partners/qiskit_ibm_provider/>
```

### CODE REVIEW

**Comment by HuangJunye** [comment]
#1638 is merged, removing `ON HOLD`.

File: docs/index.rst [hunk 1]
```diff
@@ -140,4 +138,5 @@ Interested in quantum hardware design?
    :hidden:
 
    Qiskit Aer <apidoc/aer>
-   Qiskit IBM Quantum Provider <apidoc/ibmq-provider>
+   Qiskit IBM Quantum Provider (deprecated) <apidoc/ibmq-provider>
+   Qiskit IBM Provider (new) <https://qiskit.org/documentation/partners/qiskit_ibm_provider/>
```
Comment (1ucian0) [inline docs/index.rst]:
```suggestion
   Qiskit IBM Quantum Provider (deprecated) <apidoc/ibmq-provider>
```

**Comment by HuangJunye** [comment]
On hold until https://github.com/Qiskit/qiskit-ibmq-provider/pull/1156 is merged and made into a new release of `qiskit-ibmq-provider`.

**Comment by HuangJunye** [comment]
Can merge after https://github.com/Qiskit/qiskit/pull/1671 is merged

**Review by Eric-Arellano — APPROVED** [review]

**Pull request merged**